ner/ner_manual_annotator_training.py

```diff
@@ -0,0 +1,131 @@
+from re import L
+from typing import Any, Dict, Iterable, Optional, List, Union
+from functools import partial
+
+import spacy
+
+from prodigy.components.loaders import get_stream
+from prodigy.models.matcher import PatternMatcher
+from prodigy.core import recipe
+from prodigy.types import RecipeSettingsType, TaskType
+from prodigy.util import split_string, get_labels, msg, log, INPUT_HASH_ATTR
+from prodigy.components.preprocess import add_tokens
+from prodigy.recipes.ner import remove_tokens
+
+
+def spans_equal(s1: Dict[str, Any], s2: Dict[str, Any]) -> bool:
+    return s1["label"] == s2["label"] and s1["start"] == s2["start"] and s1["end"] == s2["end"]
+
+
+def ensure_span_text(eg: TaskType) -> TaskType:
+    for span in eg["spans"]:
+        if "text" not in span:
+            span["text"] = eg["text"][span["start"]:span["end"]]
+    return eg
+
+
+def validate_answer(answer: TaskType, *, known_answers: List[TaskType]):
```

Comment:
I think we should also check the label and provide some custom explanations in respect. I think the `known_answers` json could have this field in the `meta`. I'll code up a proposal on a separate branch.